### PR TITLE
feat: Add percent of total support in alerts

### PR DESCRIPTION
**Pull request by AdityaHegde**
closes [#5242](https://github.com/rilldata/rill/issues/5242)

### CODE REVIEW

File: web-common/src/features/alerts/criteria-tab/CriteriaForm.svelte [hunk 1]
```diff
@@ -35,28 +36,36 @@
...
+          return {
+            ...o,
+            label: `${o.label} ${comparisonLabel}`,
+          };
+        })
+      : MeasureFilterBaseTypeOptions),
+    ...(selectedMeasure?.validPercentOfTotal
+      ? [MeasureFilterPercentOfTotalOption]
+      : []),
+  ];
```
Comment (ericpgreen2) [inline web-common/src/features/alerts/criteria-tab/CriteriaForm.svelte:66]: This is pretty hairy. A function `makeTypeOptions(...)` could be warranted.

**Review by ericpgreen2 — COMMENTED** [review]
UXQA –
1. Per Alex's requirements, we should validate that the input is within 0 and 100, otherwise show a validation error message.
    image
2. I don't think it makes sense to show "% of total" when the measure is not split by a dimension. It'll always be 100%, right? @mindspank, you say in the requirements "Percent of total is independent of comparison and should always be able available". Am I missing something?
    image

File: web-common/src/features/alerts/criteria-tab/parseCriteriaError.ts [hunk 2]
```diff
@@ -6,5 +6,8 @@ export function parseCriteriaError(errStr: string, index: number): string {
   const match = criteriaParserRegex.exec(errStr);
   if (!match) return "";
   const [, matchedIndex, matchedErr] = match;
-  return Number(matchedIndex) === index ? matchedErr : "";
+  return Number(matchedIndex) === index
+    ? // `value1` in error is not user friendly. replacing with `criteria value`
```
Comment (ericpgreen2) [inline web-common/src/features/alerts/criteria-tab/parseCriteriaError.ts:10]: Food for thought for later: I was confused myself when I saw "value1". I wonder if there's a way we can restructure the `MeasureFilterEntry` type to make it more self-explanatory.

**Review by ericpgreen2 — APPROVED** [review]

File: web-common/src/features/alerts/criteria-tab/parseCriteriaError.ts (hunk 2 above)
Comment (AdityaHegde) [reply web-common/src/features/alerts/criteria-tab/parseCriteriaError.ts:10]: Ya. When we redesign measure filters in dashboard we can get rid of `value1` and `value2`

**Pull request merged**